src/main/java/org/spongepowered/api/text/transform/TextFormatter.java

```diff
@@ -0,0 +1,312 @@
+        for (int i = index; i < size(); i++) {
+            E e = get(i);
+            if (clazz.isAssignableFrom(e.getClass())) {
+                return Optional.of((T) e);
+            }
+        }
+        return Optional.empty();
+    }
+
+    /**
+     * Returns the first element of the specified type.
+     *
+     * @param clazz Type to get
+     * @param <T> Type of {@link TextRepresentable}
+     * @return TextRepresentable of type if found, empty otherwise
+     */
+    default <T extends TextRepresentable> Optional<T> first(Class<T> clazz) {
+        return firstAfter(0, clazz);
+    }
+
+    /**
+     * Applies the specified consumer to each element of the specified type
+     * after the specified index.
+     *
+     * @param index Index to start at
+     * @param clazz Class of type
+     * @param consumer Consumer
+     * @param <T> Type
+     */
+    @SuppressWarnings("unchecked")
+    default <T extends TextRepresentable> void forEachAfter(int index, Class<T> clazz, Consumer<T> consumer) {
+        checkNotNull(clazz, "class");
+        for (int i = index; i < size(); i++) {
+            E e = get(i);
+            if (clazz.isAssignableFrom(e.getClass())) {
+                consumer.accept((T) e);
+            }
+        }
+    }
+
+    /**
+     * Applies the specified consumer to each element of the specified type
+     * after the specified index.
+     *
+     * @param clazz Class of type
+     * @param consumer Consumer
+     * @param <T> Type
+     */
+    default <T extends TextRepresentable> void forEach(Class<T> clazz, Consumer<T> consumer) {
+        forEachAfter(0, clazz, consumer);
+    }
+
+    /**
+     * Replaces the element previously at the specified index with the
+     * specified element.
+     *
+     * @param i Index to replace
+     * @param e Element to replace with
+     * @return Element previously at index
+     */
+    E set(int i, E e);
+
+    /**
+     * Returns the amount of elements in this formatter.
+     *
+     * @return Amount of elements
+     */
+    int size();
+
+    /**
+     * Returns true if the formatter contains no elements.
+     *
+     * @return True if contains no elements
+     */
+    boolean isEmpty();
+
+    /**
+     * Returns true if the specified element is in the formatter.
+     *
+     * @param e Element to check
+     * @return True if in formatter
+     */
+    boolean contains(E e);
+
+    /**
+     * Clears all elements from this formatter.
+     */
+    void clear();
+
+    /**
+     * Adds the specified element to the end of this formatter.
+     *
+     * @param element Element to add
+     * @return True if the formatter changed as a result of the call
+     */
+    boolean add(E element);
+
+    /**
+     * Adds the specified elements to the end of this formatter.
+     *
+     * @param elements Elements to add
+     * @return True if the formatter changed as a result of the call
+     */
+    boolean add(Collection<E> elements);
```

Comment:
`List.addAll` takes a `Collection` so if we have an `Iterable` we need to create a new `Collection from it i.e `Lists.newArrayList(iterable)`,